### PR TITLE
Update template example

### PR DESCRIPTION
**Pull request by lowecg**
as discussed, here's an updated template `core.cljc` file

The interceptor comment may be a bit wordy and better served by referencing a doc page once we have one

### CODE REVIEW

File: modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc [hunk 1]
```diff
@@ -1,18 +1,62 @@
-(ns {{main-ns}}.core
+(ns {{main-ns}} .core
```
Comment (FieryCod) [inline modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc:1]: Please remove this extra whitespace.

File: modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc [hunk 2]
```diff
@@ -1,18 +1,62 @@
...
+
+(i/definterceptor LogIncomingRequest
+                  {:enter (fn [request]
+                            (println request)
+                            request)})
+
+(i/definterceptor AddHeaderToResponse
+                  {:leave (fn [response]
+                            (hr/header response "Custom-Header" "Some Value"))})
+
```
Comment (FieryCod) [inline modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc:42]: Please remove those extra two lines

File: modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc [hunk 3]
```diff
@@ -1,18 +1,62 @@
...
+             (hr/text (say-hello)))
+
+
+;; (native only) specify the Lambda's entry point as a static main function when generating a class file
 (native/entrypoint [#'ExampleLambda])
 
+
+;; (native only) Executes the body in a safe agent context for native configuration generation.
+;; Useful when it's hard for agent payloads to cover all logic branches.
+
```
Comment (FieryCod) [inline modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc:60]: Please remove this blank line

File: modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc [hunk 4]
```diff
@@ -1,18 +1,62 @@
...
+
+(h/deflambda ExampleLambda
+             "I can run on Java, Babashka or Native runtime..."
+             < {:interceptors [LogIncomingRequest AddHeaderToResponse]}
+             [{:keys [event ctx] :as request}]
+
+             ;; return a successful plain text response. See also, hr/json
+             (hr/text (say-hello)))
+
```
Comment (FieryCod) [inline modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc:52]: Please remove those 2  blank lines

File: modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc [hunk 5]
```diff
@@ -1,18 +1,62 @@
...
+;; An interceptor is a pair of unary functions representing inbound requests and outbound responses.
+;;
+;; holy-lambda calls the :enter function on the way "in" to handling a request. It calls the :leave function on the way back "out".
+;;
+;; The :enter function is called with the request map and must return either a request map or a channel that will deliver a request map.
+;; Similarly, the :leave function is called with the response map and must return either a response map or a channel that will deliver a response map.
+;;
+;; Interceptors may be chained together:
+;;     < {:interceptors [Interceptor1 Interceptor2]}
+;;
```
Comment (FieryCod) [inline modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc:27]: There is also another way of joining interceptors together.
```
(def mixin1 {:interceptors [interceptor1]})
(def mixin2 {:interceptors [interceptor2]})

(def mixin3 (h/merge-mixins mixin1 mixin2)) ;; => {:interceptors [interceptor1 interceptor2]}
```

Comment (lowecg) [reply modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc]: added to docs

File: modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc [hunk 6]
```diff
@@ -1,18 +1,62 @@
...
+  []
+  #?(:bb  (str "Hello world. Babashka is a sweet friend of mine! Babashka version: " (System/getProperty "babashka.version"))
+     :clj "Hello world"))
+
+
+;; An interceptor is a pair of unary functions representing inbound requests and outbound responses.
+;;
+;; holy-lambda calls the :enter function on the way "in" to handling a request. It calls the :leave function on the way back "out".
+;;
+;; The :enter function is called with the request map and must return either a request map or a channel that will deliver a request map.
```
[the remark this answers is not in the document]
Comment (lowecg) [reply modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc]: added to docs

File: doc/2-01-interceptors.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+Either the `:enter` or `:leave` function may be omitted without harm.
+
+Logically speaking, interceptors form a _stack_. All the `:enter` functions are called in order. Each one receives the **request** map and returns a (possibly modified) request map. Once all the interceptors have been called, the **handler** is invoked and will process the request and emit a response. The resulting **response** map gets passed through the interceptors' `:leave` functions, but in reverse order, as shown here:
+
+![interceptor-call-chain](images/interceptor-call-chain.png "Execution call chain with interceptors")
+
+Interceptors may be chained together:
+```clojure
+(h/deflambda ExampleLambda
```
Comment (FieryCod) [inline doc/2-01-interceptors.md:32]: The formatting seems wrong. Deflambda should be formatted as defn. https://cursive-ide.com/userguide/formatting.html#code-style-settings

File: modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc [hunk 8]
```diff
@@ -1,18 +1,47 @@
...
+  []
+  #?(:bb  (str "Hello world. Babashka is a sweet friend of mine! Babashka version: " (System/getProperty "babashka.version"))
+     :clj "Hello world"))
 
+;; (alpha feature) An interceptor is a single or pair of unary functions representing inbound requests and/or outbound
+;; responses, allowing code to inspect or amend requests/responses between AWS and your Lambda function.
+;; See here https://cljdoc.org/d/fierycod/holy-lambda/CURRENT/doc/features/interceptors
+
+(i/definterceptor LambdaLogger
+                  {:enter (fn [request]
```
Comment (FieryCod) [inline modules/holy-lambda-template/resources/leiningen/new/holy_lambda/core.cljc:21]: Formatting seems wrong

**Review by FieryCod — COMMENTED** [review]
There are some small formatting issues. Probably due to incorrect intelij config. Everyhing else seems rock solid!
